packages/instant/src/components/buy_button.tsx

```diff
@@ -42,10 +46,18 @@ export class BuyButton extends React.Component<BuyButtonProps> {
             return;
         }
 
+        this.props.onPendingValidation(buyQuote);
+        const takerAddress = await getBestAddress();
+
+        const hasSufficientFunds = await balanceUtil.hasSufficientFunds(takerAddress, buyQuote, web3Wrapper);
+        if (!hasSufficientFunds) {
+            this.props.onValidationFail(buyQuote, ZeroExInstantError.InsufficientBalance);
+            return;
+        }
+
         let txHash: string | undefined;
-        this.props.onAwaitingSignature(buyQuote);
         try {
-            txHash = await assetBuyer.executeBuyQuoteAsync(buyQuote);
+            txHash = await assetBuyer.executeBuyQuoteAsync(buyQuote, { takerAddress });
         } catch (e) {
```

Comment:
@BMillman19 I think you mentioned there could be validation errors from `executeBuyQuoteAsync`, is that correct?  If so -- we should we identify them and send them to onValidationFail?

Reply:
yes, it is `AssetBuyerError.TransactionValueTooLow`

Reply:
If this pops up it can mean quote is stale so we what a slightly different behavior

Reply:
OK, will treat this as a validation fail for now